### PR TITLE
TF-1014: [Dev] Show quota at the bottom of mailbox

### PR DESCRIPTION
**Pull request by ManhNTX**
Screen Shot 2022-11-10 at 10 29 25
Screen Shot 2022-11-10 at 10 29 20
Screen Shot 2022-11-09 at 16 45 00
Screen Shot 2022-11-10 at 10 29 32
Screen Shot 2022-11-10 at 10 29 39

### CODE REVIEW

**Comment by chibenwa** [comment]
40% seems low for an alert. I would prefer to get the alert at 80% or even 90%.

**Comment by hoangdat** [comment]
Please follow the discussion in https://www.figma.com/file/rRVxJpCu1G22D0jqyvK7I9/TeamMail?node-id=6475%3A281844

File: lib/main/localizations/app_localizations.dart [hunk 1]
```diff
@@ -2487,4 +2487,34 @@ class AppLocalizations {
...
+    return Intl.message(
+      'You are running out of storage ($progress%).',
+      name: 'textQuotasWarningTitle',
+      args: [progress],
+    );
+  }
+
+  String get textQuotasWarningContent {
+    return Intl.message(
+      "Soon you won't be able to email in Tmail. Please clean your storage or upgrade your storage to get full features in Tmail.",
```
Comment (hoangdat) [inline lib/main/localizations/app_localizations.dart:2516]: Please note that, from right now, it should be `Team Mail`, some designs are not up-to-date, but please change it during your work

File: lib/features/mailbox/presentation/mailbox_view.dart [hunk 2]
```diff
@@ -84,7 +85,7 @@ class MailboxView extends GetWidget<MailboxController> {
                       } else {
                         return const SizedBox.shrink();
                       }
-                    })
+                    }),
```
Comment (dab246) [inline lib/features/mailbox/presentation/mailbox_view.dart:88]: IMO, It's not needed here.

File: lib/features/quotas/data/network/quotas_api.dart [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        JmapRequestBuilder(_httpClient, ProcessingInvocation());
+    final getQuotaMethod = GetQuotaMethod(accountId);
+    final getQuotaInvocation = requestBuilder.invocation(getQuotaMethod);
+    final response = await (requestBuilder
+          ..usings(getQuotaMethod.requiredCapabilities))
+        .build()
+        .execute();
+
+    final getQuotaResponse = response.parse<GetQuotaResponse>(
+        getQuotaInvocation.methodCallId, GetQuotaResponse.deserialize);
```
Comment (dab246) [inline lib/features/quotas/data/network/quotas_api.dart:25]: Each parameter should be on 1 line

File: lib/features/quotas/presentation/quotas_controller.dart [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        consumeState(_getQuotasInteractor.execute(mailboxDashBoardController.accountId.value!));
+      }
+    });
+
+  }
+
+  @override
+  void onDone() {
+    viewState.value.fold(
+            (failure) {
```
Comment (dab246) [inline lib/features/quotas/presentation/quotas_controller.dart:46]: remove 1 indent

File: lib/features/quotas/presentation/quotas_controller.dart [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+          if (failure is GetQuotasFailure) {
+            logError('QuotasController::onDone():[GetQuotasFailure]: ${failure.exception}');
+          }
+        },
+            (success) {
```
Comment (dab246) [inline lib/features/quotas/presentation/quotas_controller.dart:51]: same

**Review by dab246 — APPROVED** [review]
lgtm

**Pull request merged**